healthcheck/healthcheck_test.go

```diff
@@ -2,95 +2,24 @@ package healthcheck
 
 import (
 	"context"
-	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/containous/traefik/testhelpers"
 	"github.com/vulcand/oxy/roundrobin"
 )
 
 const healthCheckInterval = 100 * time.Millisecond
 
-type testLoadBalancer struct {
```

Comment:
> you _do_ need to know the implementation details in order to understand the test

It's a bad test for me.
A test must clear, readable and understandable by itself.

My main motivation: test readability, always the readability 😃